lib/std/math/fabs.zig

```diff
@@ -16,86 +15,28 @@ const maxInt = std.math.maxInt;
 ///  - fabs(nan)   = nan
 pub fn fabs(x: anytype) @TypeOf(x) {
     const T = @TypeOf(x);
-    return switch (T) {
-        f16 => fabs16(x),
-        f32 => fabs32(x),
-        f64 => fabs64(x),
-        f128 => fabs128(x),
-        else => @compileError("fabs not implemented for " ++ @typeName(T)),
-    };
-}
-
-fn fabs16(x: f16) f16 {
-    var u = @bitCast(u16, x);
-    u &= 0x7FFF;
-    return @bitCast(f16, u);
-}
-
-fn fabs32(x: f32) f32 {
-    var u = @bitCast(u32, x);
-    u &= 0x7FFFFFFF;
-    return @bitCast(f32, u);
-}
-
-fn fabs64(x: f64) f64 {
-    var u = @bitCast(u64, x);
-    u &= maxInt(u64) >> 1;
-    return @bitCast(f64, u);
-}
+    const IntT = std.meta.Int(.unsigned, @bitSizeOf(T));
+    if (@typeInfo(T) != .Float) {
+        @compileError("fabs not implemented for " ++ @typeName(T));
+    }
 
-fn fabs128(x: f128) f128 {
-    var u = @bitCast(u128, x);
-    u &= maxInt(u128) >> 1;
-    return @bitCast(f128, u);
+    const remove_sign = @as(IntT, math.maxInt(IntT)) >> 1;
+    return @bitCast(T, @bitCast(IntT, x) & remove_sign);
 }
 
 test "math.fabs" {
-    try expect(fabs(@as(f16, 1.0)) == fabs16(1.0));
-    try expect(fabs(@as(f32, 1.0)) == fabs32(1.0));
-    try expect(fabs(@as(f64, 1.0)) == fabs64(1.0));
-    try expect(fabs(@as(f128, 1.0)) == fabs128(1.0));
-}
-
-test "math.fabs16" {
-    try expect(fabs16(1.0) == 1.0);
-    try expect(fabs16(-1.0) == 1.0);
-}
-
-test "math.fabs32" {
-    try expect(fabs32(1.0) == 1.0);
-    try expect(fabs32(-1.0) == 1.0);
-}
-
-test "math.fabs64" {
-    try expect(fabs64(1.0) == 1.0);
-    try expect(fabs64(-1.0) == 1.0);
-}
-
-test "math.fabs128" {
-    try expect(fabs128(1.0) == 1.0);
-    try expect(fabs128(-1.0) == 1.0);
-}
```

Comment:
you did not convert these to comptime and did not make an argument why.

Reply:
These were removed because those individual functions do not exist anymore, the function returns `@TypeOf(x)` and doesn't need to be tested for a `switch (@TypeOf(x))` working now. Unless I misunderstand what you're asking.